tests/ga/test_extension.py

```diff
@@ -1513,26 +1475,129 @@ def test_upgrade_failure_with_exception_handling(self, patch_get_update_command,
         """
         Extension upgrade failure should not be retried
         """
-        test_data = WireProtocolData(DATA_FILE_EXT_SINGLE)
-        exthandlers_handler, protocol = self._create_mock(test_data, *args)
+        test_data, exthandlers_handler, protocol = self._setup_update_test_and_update_gs(patch_get_update_command,
+                                                                                         *args)
 
-        # Ensure initial install and enable is successful
         exthandlers_handler.run()
-        self.assertEqual(0, patch_get_update_command.call_count)
+        self.assertEqual(1, patch_get_update_command.call_count)
+        self.assertEqual(1, patch_handle_ext_handler_error.call_count)
 
-        self._assert_handler_status(protocol.report_vm_status, "Ready", expected_ext_count=1, version="1.0.0")
-        self._assert_ext_status(protocol.report_ext_status, "success", 0)
+    @patch('azurelinuxagent.ga.exthandlers.HandlerManifest.get_disable_command')
+    def test_extension_upgrade_should_pass_when_continue_on_update_failure_is_true_and_prev_version_disable_fails(
+            self, patch_get_disable_command, *args):
+        test_data, exthandlers_handler, protocol = self._setup_update_test_and_update_gs(patch_get_disable_command,
+                                                                                         *args)
 
-        # Next incarnation, update version
-        test_data.goal_state = test_data.goal_state.replace("<Incarnation>1<", "<Incarnation>2<")
-        test_data.ext_conf = test_data.ext_conf.replace('version="1.0.0"', 'version="1.0.1"')
-        test_data.manifest = test_data.manifest.replace('1.0.0', '1.0.1')
+        with patch('azurelinuxagent.ga.exthandlers.HandlerManifest.is_continue_on_update_failure', return_value=True) \
+                as mock_continue_on_update_failure:
+            exthandlers_handler.run()
+            self.assertEqual(1, patch_get_disable_command.call_count)
+            self.assertEqual(2, mock_continue_on_update_failure.call_count,
+                             "This should be called twice, for both disable and uninstall")
 
-        # Update command should fail
-        patch_get_update_command.return_value = "exit 1"
-        exthandlers_handler.run()
-        self.assertEqual(1, patch_get_update_command.call_count)
-        self.assertEqual(1, patch_handle_ext_handler_error.call_count)
+        self._assert_handler_status(protocol.report_vm_status, "Ready", expected_ext_count=1, version="1.0.1")
+
+    @patch('azurelinuxagent.ga.exthandlers.HandlerManifest.is_continue_on_update_failure', return_value=True)
+    @patch('azurelinuxagent.ga.exthandlers.HandlerManifest.get_uninstall_command')
+    def test_extension_upgrade_should_pass_when_continue_on_update_failue_is_true_and_prev_version_uninstall_fails(
```

Comment:
nit: failue -> failure

Reply:
Actually, there are several occurrences of this typo. Search and replace :)

Reply:
Haha thanks done!